{"seeAlsoSections":[{"anchor":"Errors","title":"Errors","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/HeaderDecodingError"],"generated":true}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["macOS"],"tokens":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"AWSServerError"}],"languages":["swift"]}]}],"sections":[],"relationshipsSections":[{"kind":"relationships","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","doc:\/\/codes.soto.soto-core\/SQ","doc:\/\/codes.soto.soto-core\/s5ErrorP","doc:\/\/codes.soto.soto-core\/s8SendableP"],"title":"Conforms To","type":"conformsTo"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/sotocore\/awsservererror"]}],"kind":"symbol","topicSections":[{"title":"Initializers","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/init(errorCode:context:)"],"generated":true,"anchor":"Initializers"},{"anchor":"Instance-Properties","generated":true,"identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/context","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/errorCode"],"title":"Instance Properties"},{"title":"Type Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/internalFailure","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/serviceUnavailable"],"generated":true,"anchor":"Type-Properties"},{"title":"Default Implementations","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/AWSErrorType-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/CustomStringConvertible-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Equatable-Implementations"],"generated":true,"anchor":"Default-Implementations"}],"identifier":{"url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","interfaceLanguage":"swift"},"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore"]]},"metadata":{"modules":[{"name":"SotoCore"}],"symbolKind":"struct","roleHeading":"Structure","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"AWSServerError","kind":"identifier"}],"title":"AWSServerError","role":"symbol","externalID":"s:8SotoCore14AWSServerErrorV","navigatorTitle":[{"text":"AWSServerError","kind":"identifier"}]},"schemaVersion":{"minor":3,"major":0,"patch":0},"references":{"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError":{"title":"AWSServerError","fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSServerError"}],"url":"\/documentation\/sotocore\/awsservererror","role":"symbol","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","type":"topic","kind":"symbol","navigatorTitle":[{"text":"AWSServerError","kind":"identifier"}]},"doc://codes.soto.soto-core/documentation/SotoCore/AWSRawError":{"role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","title":"AWSRawError","abstract":[{"text":"Raw unprocessed error.","type":"text"}],"fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSRawError"}],"type":"topic","kind":"symbol","url":"\/documentation\/sotocore\/awsrawerror","navigatorTitle":[{"text":"AWSRawError","kind":"identifier"}]},"doc://codes.soto.soto-core/SQ":{"type":"unresolvable","identifier":"doc:\/\/codes.soto.soto-core\/SQ","title":"Swift.Equatable"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorContext":{"url":"\/documentation\/sotocore\/awserrorcontext","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","kind":"symbol","abstract":[{"text":"Additional information about error","type":"text"}],"role":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"text":"AWSErrorContext","kind":"identifier"}],"title":"AWSErrorContext","type":"topic","navigatorTitle":[{"text":"AWSErrorContext","kind":"identifier"}]},"doc://codes.soto.soto-core/s5ErrorP":{"type":"unresolvable","identifier":"doc:\/\/codes.soto.soto-core\/s5ErrorP","title":"Swift.Error"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/errorCode":{"title":"errorCode","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"errorCode"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"url":"\/documentation\/sotocore\/awsservererror\/errorcode","role":"symbol","abstract":[{"text":"return error code string","type":"text"}],"kind":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/errorCode"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/internalFailure":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/internalFailure","title":"internalFailure","fragments":[{"text":"static","kind":"keyword"},{"text":" ","kind":"text"},{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"internalFailure"},{"text":": ","kind":"text"},{"text":"AWSServerError","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore14AWSServerErrorV"}],"type":"topic","abstract":[],"url":"\/documentation\/sotocore\/awsservererror\/internalfailure","role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/serviceUnavailable":{"role":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/serviceunavailable","abstract":[],"fragments":[{"text":"static","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"text":"serviceUnavailable","kind":"identifier"},{"text":": ","kind":"text"},{"text":"AWSServerError","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore14AWSServerErrorV"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/serviceUnavailable","title":"serviceUnavailable","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/HeaderDecodingError":{"title":"HeaderDecodingError","fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"text":"HeaderDecodingError","kind":"identifier"}],"url":"\/documentation\/sotocore\/headerdecodingerror","role":"symbol","abstract":[{"type":"text","text":"Decoding error returned when extracting data out of response headers"}],"kind":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/HeaderDecodingError","navigatorTitle":[{"kind":"identifier","text":"HeaderDecodingError"}]},"doc://codes.soto.soto-core/documentation/SotoCore":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","url":"\/documentation\/sotocore","kind":"symbol","title":"SotoCore","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"type":"topic","role":"collection"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSClientError":{"navigatorTitle":[{"text":"AWSClientError","kind":"identifier"}],"role":"symbol","title":"AWSClientError","url":"\/documentation\/sotocore\/awsclienterror","abstract":[],"kind":"symbol","type":"topic","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"AWSClientError"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/CustomStringConvertible-Implementations":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/CustomStringConvertible-Implementations","title":"CustomStringConvertible Implementations","type":"topic","abstract":[],"url":"\/documentation\/sotocore\/awsservererror\/customstringconvertible-implementations","role":"collectionGroup","kind":"article"},"doc://codes.soto.soto-core/s23CustomStringConvertibleP":{"type":"unresolvable","identifier":"doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","title":"Swift.CustomStringConvertible"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/context":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/context","title":"context","fragments":[{"text":"let","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"context"},{"kind":"text","text":": "},{"preciseIdentifier":"s:8SotoCore15AWSErrorContextV","text":"AWSErrorContext","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"type":"topic","abstract":[{"type":"text","text":"additional context information related to the error"}],"url":"\/documentation\/sotocore\/awsservererror\/context","role":"symbol","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/Equatable-Implementations":{"role":"collectionGroup","type":"topic","url":"\/documentation\/sotocore\/awsservererror\/equatable-implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Equatable-Implementations","title":"Equatable Implementations","kind":"article"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError":{"fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSResponseError","kind":"identifier"}],"navigatorTitle":[{"kind":"identifier","text":"AWSResponseError"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","kind":"symbol","url":"\/documentation\/sotocore\/awsresponseerror","type":"topic","title":"AWSResponseError","role":"symbol","abstract":[{"text":"Response Error type returned by Soto if the error code is unrecognised","type":"text"}]},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/AWSErrorType-Implementations":{"title":"AWSErrorType Implementations","kind":"article","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/AWSErrorType-Implementations","abstract":[],"url":"\/documentation\/sotocore\/awsservererror\/awserrortype-implementations","role":"collectionGroup"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType":{"type":"topic","navigatorTitle":[{"text":"AWSErrorType","kind":"identifier"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","url":"\/documentation\/sotocore\/awserrortype","kind":"symbol","title":"AWSErrorType","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorType"}],"role":"symbol","abstract":[{"text":"Standard Error type returned by Soto.","type":"text"}]},"doc://codes.soto.soto-core/s8SendableP":{"type":"unresolvable","identifier":"doc:\/\/codes.soto.soto-core\/s8SendableP","title":"Swift.Sendable"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/init(errorCode:context:)":{"title":"init(errorCode:context:)","kind":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/init(errorCode:context:)","abstract":[{"type":"text","text":"initialize AWSServerError"}],"fragments":[{"text":"init","kind":"identifier"},{"text":"?(","kind":"text"},{"kind":"externalParam","text":"errorCode"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"text":", ","kind":"text"},{"text":"context","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:8SotoCore15AWSErrorContextV","kind":"typeIdentifier","text":"AWSErrorContext"},{"text":")","kind":"text"}],"url":"\/documentation\/sotocore\/awsservererror\/init(errorcode:context:)","role":"symbol"}}}